### ISSUE TITLE
[SVGKImage initWithData:]: unrecognized selector sent

### ISSUE THREAD

**Issue by Rao27** [SVGKit/SVGKit#477]
I am trying to load SVGImage from string. For that I have written code shown below.

NSData dataForSvg = NSData.FromString (svgString.ToString());
//SVGKImage image = new SVGKImage ("svg1.svg");
SVGKImage image = new SVGKImage(dataForSvg);
_svgImgView = new SVGKFastImageView (image);

But, I am getting exception for creating SVGImage from NSData.

Exception: NSInvalidArgumentException -[SVGKImage initWithData:]: unrecognized selector sent to instance

Am I doing anything wrong here?

**Comment by adamgit** [comment]
The error literally states you are calling a method that does not exist.
You cannot invent method names and hope they exist! :)

Try reading the header file or the examples and using a valid method.

**Comment by Rao27** [comment]
I can see the list of constructor in dll file and I have copied and pasted those below. If there is a constructor which takes nsdata as argument then also the method is not present?

```
    //
    // Constructors
    //
    [Export ("initWithParsedSVG:fromSource:"), CompilerGenerated]
    public SVGKImage (SVGKParseResult parseResult, SVGKSource parseSource);

    [Export ("initWithSource:"), CompilerGenerated]
    public SVGKImage (SVGKSource source);

    [Export ("initWithData:"), CompilerGenerated]
    public SVGKImage (NSData data);

    [Export ("initWithContentsOfFile:"), CompilerGenerated]
    public SVGKImage (string path);

    [EditorBrowsable (EditorBrowsableState.Advanced), CompilerGenerated]
    protected SVGKImage (NSObjectFlag t);

    [Export ("init"), EditorBrowsable (EditorBrowsableState.Advanced), CompilerGenerated]
    public SVGKImage ();
```

**Comment by Rao27** [comment]
Okay so Constructor with SVGSource is working. I was trying to load svgimage from string. To do so, I used the code shown below.

```
        NSData dataForSvg = NSData.FromString (svgString);
        NSInputStream svgInputStream = NSInputStream.FromData (dataForSvg);
        SVGKSource svgSource = new SVGKSource (svgInputStream);

        SVGKImage image = new SVGKImage (svgSource);
        _svgImgView = new SVGKFastImageView (image);
```

**Comment by adamgit** [comment]
I hunk there's an svgksource that takes an nsstring directly, cutting out
one step.

NB: dont examine an objectiveC DLL! . Always read the header files instead.

**Comment by Rao27** [comment]
Ohh Alright. I am not sure whether its a right place to discuss further. But, nothing is coming up in to my mind so let me ask this question here, I need suggestions if there is any.

I want to interact with svg sections. Is there any way that I can interact with svg element; for example, I want to click on an element. Is it possible?

**Comment by adamgit** [comment]
Yes there are examples of this linked from the front page of the project,
and in he example app included in the project.
